### ISSUE TITLE
Incorrect LinkEvent typing

### ISSUE THREAD

**Issue by averyv** [plaid/react-native-plaid-link-sdk#326]
# The problem

Briefly describe the issue you are experiencing (or the feature you want to see added to the Plaid Link 
React Native SDK). Tell us what you were trying to do and what happened instead. 

The `LinkEvent` typing in `Types.d.ts` is as follows:
```
export interface LinkEvent {
    eventName: LinkEventName;
    metadata: LinkEventMetadata;
}
```
However, when I implement a listener to get `onEvent` callbacks as detailed [here](https://github.com/plaid/react-native-plaid-link-sdk#to-receive-onevent-callbacks), I get an object that looks like this:
```
export interface LinkEvent {
    event: string;
    metadata: LinkEventMetadata;
}
```
where the event may be "SELECT_INSTITUTION", "TRANSITION_VIEW", "OPEN", etc..

# Environment

|                        |  |
| ------------------------| ---------- |
| Plaid Link React Native          | 7.0.0                |
| ReactNative Version     | 0.63.3                |
| Occurs on Android             | not sure yet but think so                |
| Android OS Version                    | N/A                |
| Android Devices/Emulators    | N/A               |
| Occurs on iOS     | yes                |
| iOS Version           | 14.3                |
| iOS Devices/Emulators               | iPhone 11               |

# Steps to Reproduce

Implement an event listener as described [here](https://github.com/plaid/react-native-plaid-link-sdk#to-receive-onevent-callbacks) and `console.log` the `LinkEvent` object. 

# Expected Result
What did you expect to happen?

I expected the object to take this form:
```
export interface LinkEvent {
    eventName: LinkEventName;
    metadata: LinkEventMetadata;
}
```

# Screenshots
If this is a visual bug a screenshot would be helpful.

# Logs
Any relevant logs from Android Logcat or the iOS console including stacktraces.

What is logged when I `console.log` the event from my listener:
```
{"event": "SELECT_INSTITUTION", "metadata": {"errorCode": "", "errorMessage": "", "errorType": "", "exitStatus": "", "institutionId": "ins_35", "institutionName": "Chime", "institutionSearchQuery": "", "linkSessionId": "90f15fb5-5899-4e69-b992-d562fabbafa5", "metadataJson": "", "mfaType": "", "requestId": "", "timestamp": "2021-02-19T16:27:19Z", "viewName": ""}}
```

# Code To Reproduce Issue 
With sample code it's easier to reproduce the bug and it's much faster to fix it. 
Don't forget to omit any keys or personally identifiable information.
Please provide one of the following:

1. (Best) A link to a minimal demo app showing the faulty behaviour.
1. (Better) A concise code sample which can be dropped into fresh application
1. A snippet of code for your configuration for opening Link

**Comment by amytang0** [comment]
Hey thanks for catching this! I tried reproing on Android and it looks like this is an iOS only issue. We'll try to get a fix out with the next release.

Android:
```
{"eventName": "OPEN", "metadata": {"error_code": "", "error_message": "", "error_type": "", "exit_status": "", "institution_search_query": "", "link_session_id": "9f7e2cff-dee4-4333-a039-de3cefd4053c", "mfa_type": "", "request_id": "", "timestamp": "2021-02-19T20:17:08.022Z", "view_name": "CONSENT"}}
```

**Comment by averyv** [comment]
Additionally, the `LinkEventMetadata` object does not align on Android and iOS.  On Android, the fields are snake case and on iOS, the fields are all camel case. 

Android:
```
export interface LinkEvent {
    eventName: LinkEventName;
    metadata: LinkEventMetadata;
}
export interface LinkEventMetadata {
    link_session_id: string;
    mfa_type?: string;
    request_id?: string;
    view_name: LinkEventViewName;
    error_code?: string;
    error_message?: string;
    error_type?: string;
    exit_status?: string;
    institution_id?: string;
    institution_name?: string;
    institution_search_query?: string;
    timestamp: string;
}
```

iOS:
```
export interface LinkEvent {
    event: LinkEventName;
    metadata: LinkEventMetadata;
}
export interface LinkEventMetadata {
    linkSessionId: string;
    mfaType?: string;
    requestId?: string;
    viewName: LinkEventViewName;
    errorCode?: string;
    errorMessage?: string;
    errorType?: string;
    exitStatus?: string;
    institutionId?: string;
    institutionName?: string;
    institutionSearchQuery?: string;
    timestamp: string;
}
```

The `LinkError` object also does not align on iOS and Android.  On Android, the `displayMessage` field holds the error display message (and there is not a `errorDisplayMessage` field).  On iOS, there is an `errorDisplayMessage` field that holds the error display message.

**Comment by pladaub** [comment]
Looks like #329 partially addresses this. Posting on this Issue to associate the PR

**Comment by pladaub** [comment]
Fixed in 7.0.2
